packages/fxa-auth-server/lib/payments/iap/apple-app-store/subscription-purchase.ts

```diff
@@ -1,208 +1,5 @@
-    purchase.originalPurchaseDate = transactionInfo.originalPurchaseDate;
-    purchase.originalTransactionId = originalTransactionId;
-    purchase.productId = transactionInfo.productId;
-    purchase.status = subscriptionStatus;
-    purchase.type = transactionInfo.type;
-    purchase.verifiedAt = verifiedAt;
-
-    if (renewalInfo.expirationIntent) {
-      purchase.expirationIntent = renewalInfo.expirationIntent;
-    }
-    if (transactionInfo.expiresDate) {
-      purchase.expiresDate = transactionInfo.expiresDate;
-    }
-    if (renewalInfo.gracePeriodExpiresDate) {
-      purchase.gracePeriodExpiresDate = renewalInfo.gracePeriodExpiresDate;
-    }
-    if (renewalInfo.hasOwnProperty('isInBillingRetryPeriod')) {
-      // We don't check this.status === SubscriptionStatus.InBillingRetry, since
-      // it's not mutually exclusive with other subscription states (i.e.
-      // SubscriptionStatus.InBillingGracePeriod).
-      purchase.isInBillingRetry = renewalInfo.isInBillingRetryPeriod;
-    }
-    if (transactionInfo.hasOwnProperty('isUpgraded')) {
-      purchase.isUpgraded = transactionInfo.isUpgraded;
-    }
-    if (renewalInfo.offerIdentifier) {
-      purchase.offerIdentifier = renewalInfo.offerIdentifier;
-    }
-    if (renewalInfo.offerType) {
-      purchase.offerType = renewalInfo.offerType;
-    }
-    if (transactionInfo.revocationDate) {
-      purchase.revocationDate = transactionInfo.revocationDate;
-    }
-    if (transactionInfo.hasOwnProperty('revocationReason')) {
-      purchase.revocationReason = transactionInfo.revocationReason;
-    }
-
-    return purchase;
-  }
-
-  /**
-   * Converts purchase data from a Firestore document to a SubscriptionPurchase.
-   * In particular, there are some shopkeeping properties that are only used in
-   * Firestore; see FIRESTORE_OBJECT_INTERNAL_KEYS.
-   */
-  static fromFirestoreObject(firestoreObject: any) {
-    const purchase = new SubscriptionPurchase();
-    purchase.mergeWithFirestorePurchaseRecord(firestoreObject);
-    return purchase;
-  }
-
-  toFirestoreObject(): any {
-    return purchaseToFirestoreObject(this);
-  }
-
-  mergeWithFirestorePurchaseRecord(firestoreObject: any) {
-    mergePurchaseWithFirestorePurchaseRecord(this, firestoreObject);
-  }
-
-  isEntitlementActive() {
-    return [
-      SubscriptionStatus.Active,
-      SubscriptionStatus.InBillingGracePeriod,
-    ].includes(this.status);
-  }
-
-  willRenew() {
-    return this.autoRenewStatus === AutoRenewStatus.On;
-  }
-
-  isInBillingRetryPeriod() {
-    return this.isInBillingRetry;
-  }
-
-  isInGracePeriod() {
-    return this.status === SubscriptionStatus.InBillingGracePeriod;
-  }
-
-  isTestPurchase(): boolean {
-    return this.environment === Environment.Sandbox;
-  }
-
-  isFreeTrial(): boolean {
-    return this.offerType === OfferType.Introductory;
-  }
-}
+export * from 'fxa-shared/payments/iap/apple-app-store/subscription-purchase';
```

Comment:
I took this approach in an effort to minimize side effects of the refactor and to cut down on changes to existing import statements. I can also see decorating this module with auth-server only functionality down the road.

Reply:
Ah, ok, so later we could move these entirely to fxa-shared as well as the tests.